soe/resolvers.go

```diff
@@ -0,0 +1,61 @@
+package soe
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/hamba/avro/v2"
+)
+
+// SchemaResolver implementations are expected to be callable for every record,
+// so should cache expensive schema lookups.
+type SchemaResolver interface {
+	GetSchema(ctx context.Context, fingerprint []byte) (avro.Schema, error)
+}
+
+// ErrUnknownSchema is an API error that must be returned if the requested
+// schema is not known to a resolver.
+var ErrUnknownSchema = errors.New("unknown schema")
+
+// MemorySchemaRegistry is an in-memory cache, intended for testing. It is not
+// safe for multithreaded use.
+type MemorySchemaRegistry struct {
```

Comment:
You should consider (not mandated) putting the resolvers in a sub package `resolver`. This seems like a likely expansion point, and when there are a few, it would keep it all a fair bit cleaner.

Reply:
I have two concerns:

* `resolver` is a fairly non-descript and collision-prone package name, keeping it in `soe` clearly scopes it in a more unique namespace
* I didn't really expect any general reusable resolvers, since there's no standard storage implementation as far as I know (as opposed to e.g. the confluent registry wrapped under `/registry`)

But I'd be happy to move it if you think it makes sense.

Reply:
I tried moving this into `soe/resolvers` and renamed it to `MemorySchemaStore`. I still feel like it would probably be better homed in `soe`, but let me know what you think.